src/apps/backend/modules/task/internal/task-writer.ts

```diff
@@ -11,23 +11,38 @@ import TaskUtil from './task-util';
 
 export default class TaskWriter {
   public static async createTask(params: CreateTaskParams): Promise<Task> {
-    const existingTask = await TaskRepository.findOne({
-      account: params.accountId,
-      name: params.name,
-      active: true,
-    });
-    if (existingTask) {
-      throw new TaskWithNameExistsError(params.name);
-    }
-
     const createdTask = await TaskRepository.create({
       account: params.accountId,
-      name: params.name,
+      description: params.description,
+      title: params.title,
       active: true,
     });
     return TaskUtil.convertTaskDBToTask(createdTask);
   }
 
+  public static async updateTask(params: UpdateTaskParams): Promise<Task> {
+    const task = await TaskRepository.findOneAndUpdate(
+      {
+        account: params.accountId,
+        _id: params.taskId,
+        active: true,
+      },
+      {
+        $set: {
+          description: params.description,
+          title: params.title,
+        },
+      },
+      { new: true },
+    );
+
+    if (!task) {
+      throw new Error('Task not found');
```

Comment:
Please improve copyright message. There is lot of literature online about, and if you Google `how to write good error messages`, you can find some reference.

Also, I think we need to throw with appropriate HTTP status code. For example, this should probably throw 404 in case there is no task found. You may need to create a custom error message from here instead of Error